### PR TITLE
feat: allow to pass docker compose configuration as string

### PR DESCRIPTION
**Pull request by furstenheim**
Addresses this issue: https://github.com/PDMLab/docker-compose/issues/109

Probably not properly linted because of https://github.com/PDMLab/docker-compose/issues/123

### CODE REVIEW

**Comment by AlexZeitler** [comment]
@furstenheim Thanks for your contribution! Would you mind adding a short description to the `readme.md` for this new feature?

**Review by AlexZeitler — APPROVED** [review]

**Comment by furstenheim** [comment]
Sure, no problem

File: test/index.test.ts [hunk 1]
```diff
@@ -302,6 +302,27 @@ test('ensure run and exec with command defined as array are working', async ():
   await compose.down({ cwd: path.join(__dirname), log: logOutput });
 });
 
+test('build accepts config as string', async (): Promise<void> => {
+  const configuration = await new Promise(function (resolve, reject) {
```
Comment (Steveb-p) [inline test/index.test.ts:306]: This should allow typescript to properly type-check the `resolve` call and prevent the need for typecasting later on in this code.
```suggestion
  const configuration = await new Promise<string>(function (resolve, reject) {
```

File: test/index.test.ts [hunk 2]
```diff
@@ -302,6 +302,27 @@ test('ensure run and exec with command defined as array are working', async ():
...
+    readFile(path.join(__dirname, 'docker-compose-2.yml'), function (err, content) {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(content.toString());
+    })
+  });
+  const config = {
+    configAsString: <string> configuration,
+    log: logOutput
```
Comment (Steveb-p) [inline test/index.test.ts:316]: nitpick:
```suggestion
    log: logOutput,
```

File: test/index.test.ts [hunk 3]
```diff
@@ -302,6 +302,27 @@ test('ensure run and exec with command defined as array are working', async ():
...
 
+test('build accepts config as string', async (): Promise<void> => {
+  const configuration = await new Promise(function (resolve, reject) {
+    readFile(path.join(__dirname, 'docker-compose-2.yml'), function (err, content) {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(content.toString());
+    })
+  });
```
Comment (Steveb-p) [inline test/index.test.ts:313]: Promise resolution function does not actually return anything (at least it shouldn't).
```suggestion
  const configuration = await new Promise(function (resolve, reject): void {
    readFile(path.join(__dirname, 'docker-compose-2.yml'), function (err, content) {
      if (err) {
        reject(err);
      }
      resolve(content.toString());
    })
  });
```

**Comment by furstenheim** [comment]
I've uploaded a new version with the discussed fixes

File: test/index.test.ts [hunk 4]
```diff
@@ -302,6 +302,28 @@ test('ensure run and exec with command defined as array are working', async ():
   await compose.down({ cwd: path.join(__dirname), log: logOutput });
 });
 
+test('build accepts config as string', async (): Promise<void> => {
+  const configuration = await new Promise<string>(function (resolve, reject) {
```
Comment (Steveb-p) [inline test/index.test.ts:306]:
```suggestion
  const configuration = await new Promise<string>(function (resolve, reject): void {
```

**Review by Steveb-p — APPROVED** [review]

**Comment by AlexZeitler** [comment]
Thanks @furstenheim @Steveb-p 📦  `docker-compose@0.23.6` is available on npm now.

**Pull request merged**